### PR TITLE
Anko - Change Splash Screen from XML To Anko Layout

### PR DESCRIPTION
**Pull request by at-toannguyen1**
#### Related issues (Required)
- Task #107 
- Bug #

#### Description (Required)
- [x] Change Splash Screen from  XML to Anko layout

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                }.lparams(matchParent, matchParent)
+                imageView {
+                    backgroundResource = R.drawable.bg_map
+                }.lparams(matchParent, matchParent)
+            }
+            verticalLayout {
+                gravity = Gravity.CENTER_HORIZONTAL
+                imageView {
+                    backgroundResource = R.drawable.ic_location_round
+                }.lparams(dip(250), dip(250)) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    topMargin = dip(30)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                }
+                textView(R.string.splash_name_application) {
+                    textSize = px2dip(dimen(R.dimen.splash_size_tv_app_name))
+                    typeface = Typeface.DEFAULT_BOLD
+                    textColor = Color.BLACK
+                }.lparams(wrapContent, wrapContent) {
+                    topMargin = dip(30)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                }
+                textView(R.string.splash_description_app) {
+                    textColor = Color.BLACK
+                    textSize = px2dip(dimen(R.dimen.splash_tv_size_app_description))
+                }.lparams(wrapContent, wrapContent) {
+                    topMargin = dip(40)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                }.lparams(wrapContent, wrapContent) {
+                    bottomMargin = dimen(R.dimen.splash_progressBar_margin_bottom)
+                }
+                button(R.string.splash_button_enable_location) {
+                    backgroundResource = R.drawable.common_bg_button_enable_location
+                    leftPadding = dimen(R.dimen.splash_buttonEnableButton_padding)
+                    rightPadding = dimen(R.dimen.splash_buttonEnableButton_padding)
+                    textColor = ContextCompat.getColor(context, R.color.colorWhite)
+                }.lparams(wrapContent, wrapContent) {
+                    topMargin = dip(30)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import android.util.DisplayMetrics
+import android.view.WindowManager
+
+/**
+ * ScreenUtil.
+ *
+ * @author at-ToanNguyen
+ */
+object ScreenUtil {
+    fun getWidthScreen(context: Context): Int {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        val wm = context
+                .getSystemService(Context.WINDOW_SERVICE) as WindowManager
+        val dimension = DisplayMetrics()
+        wm.defaultDisplay.getMetrics(dimension)
+        return dimension.widthPixels
+    }
+
+    fun getHeightScreen(context: Context): Int {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import android.view.WindowManager
+
+/**
+ * ScreenUtil.
+ *
+ * @author at-ToanNguyen
+ */
+object ScreenUtil {
+    fun getWidthScreen(context: Context): Int {
+        val wm = context
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+                .getSystemService(Context.WINDOW_SERVICE) as WindowManager
+        val dimension = DisplayMetrics()
+        wm.defaultDisplay.getMetrics(dimension)
+        return dimension.widthPixels
+    }
+
+    fun getHeightScreen(context: Context): Int {
+        val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+        val dimension = DisplayMetrics()
+        wm.defaultDisplay.getMetrics(dimension)
+        return dimension.heightPixels
+    }
+}
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: WARNING: Checks whether files end with a line separator.

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+     * @param context is current context
+     * @return return height screen in pixel
+     */
+    fun getHeightScreen(context: Context): Int {
+        val wm = context.getSystemService(Context.WINDOW_SERVICE) as? WindowManager
+        val dimension = DisplayMetrics()
+        wm?.defaultDisplay?.getMetrics(dimension)
+        return dimension.heightPixels
+    }
+}
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: WARNING: Checks whether files end with a line separator.

File: app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    internal lateinit var imgBehindBackground: ImageView
+    internal lateinit var imgCircle: ImageView
+    internal lateinit var tvAppDescription: TextView
+    internal lateinit var progressBar: ProgressBar
+    internal lateinit var btnEnableLocation: Button
+    override fun createView(ui: AnkoContext<SplashActivity>): View = with(ui) {
+        scrollView {
+            relativeLayout {
+                frameLayout {
+                    lparams(ScreenUtil.getWidthScreen(context), ScreenUtil.getHeightScreen(context))
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt]: change line to `lparams(matchParent, matchParent)` also check all file

File: app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+                        topMargin = dimen(R.dimen.splash_img_location_margin_top)
+                    }
+                    textView(R.string.splash_name_application) {
+                        textSize = px2dip(dimen(R.dimen.splash_size_tv_app_name))
+                        typeface = Typeface.DEFAULT_BOLD
+                        textColor = Color.BLACK
+                    }.lparams(wrapContent, wrapContent) {
+                        topMargin = dimen(R.dimen.splash_tv_app_name_margin_top)
+                    }
+                    tvAppDescription = textView(R.string.splash_description_app) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivityUI.kt]: Please compare with main design

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finished review this pull request.
Please check it again, thank you!

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 14]
```diff
@@ -0,0 +1,41 @@
+package vn.asiantech.way.utils
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: Remove this file

**Comment by at-tienhoang** [comment]
Add this layout into Activity

File: app/src/main/java/vn/asiantech/way/ui/splash/SplashActivity.kt [hunk 15]
```diff
@@ -56,7 +56,8 @@ class SplashActivity : BaseActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_splash)
+        ui=SplashActivityUI()
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/splash/SplashActivity.kt]: WARNING:

**Pull request merged**